File: src/lib/api/automation.ts
import { apiFetch } from './client';

export type SequenceStatus = 'draft' | 'active' | 'paused';

export interface SequenceStepDTO {
  id: string;
  step_order: number;
  delay_hours: number;
  subject: string;
  body: string;
}

export interface SequenceDTO {
  id: string;
  workspace_id: string;
  name: string;
  trigger: string;
  status: SequenceStatus;
  created_at: string;
  steps: SequenceStepDTO[];
}

export interface SequenceListResponseDTO {
  items: SequenceDTO[];
}

export function listSequences(token: string | null) {
  return apiFetch<SequenceListResponseDTO>('/api/v1/automation/sequences', token);
}

export interface CreateSequenceStepInput {
  subject: string;
  body: string;
  delay_hours?: number;
}

export interface CreateSequenceInput {
  name: string;
  trigger?: string;
  steps: CreateSequenceStepInput[];
}

export function createSequence(token: string | null, input: CreateSequenceInput) {
  return apiFetch<SequenceDTO>('/api/v1/automation/sequences', token, {
    method: 'POST',
    body: JSON.stringify(input)
  });
}

export function updateSequenceStatus(token: string | null, sequenceId: string, status: SequenceStatus) {
  return apiFetch<SequenceDTO>(`/api/v1/automation/sequences/${sequenceId}/status`, token, {
    method: 'PATCH',
    body: JSON.stringify({ status })
  });
}

export function deleteSequence(token: string | null, sequenceId: string) {
  return apiFetch<undefined>(`/api/v1/automation/sequences/${sequenceId}`, token, {
    method: 'DELETE'
  });
}

File: backend/app/schemas/automation.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field

SequenceStatus = Literal["draft", "active", "paused"]


class SequenceStepCreate(BaseModel):
    subject: str
    body: str
    delay_hours: int = Field(default=0, ge=0)


class SequenceStepRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    step_order: int
    delay_hours: int
    subject: str
    body: str


class SequenceCreate(BaseModel):
    name: str
    trigger: str = "lead_created"
    steps: list[SequenceStepCreate] = []


class SequenceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    workspace_id: str
    name: str
    trigger: str
    status: SequenceStatus
    created_at: datetime
    steps: list[SequenceStepRead]


class SequenceListResponse(BaseModel):
    items: list[SequenceRead]


class SequenceStatusUpdate(BaseModel):
    status: SequenceStatus


class InboundLeadCapture(BaseModel):
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    source: Optional[str] = "webhook"
